### PR TITLE
#61 Added default item in case of empty list in labels dropdown in subscriber section

### LINKED ISSUE #61: 🤔 [ISSUE] - in case of no labels a disables no lable found item should be present in place of just an empty dropdown list
### Description

currently in the subscriber module if the labels list is empty then an empty dropdown is shown that looks bad and is not a good UI experience.

in place of this empty dropdown a item should be present there with the title like 'No label found' which is not selectabel.

Screenshot from 2024-09-22 12-01-18

### Reproduction steps

```bash
1. Go to 'Subscribers'
2. Click on 'labels'
3. make sure the labels list is empty
4. a empty dropdown is present.
```

### Logs

_No response_

### Browsers

_No response_

### OS

_No response_

### PR DESCRIPTION
**Pull request by JaiSharma1710**
# Motivation
currently in the subscriber module if the labels list is empty then an empty dropdown is shown that looks bad and is not a good UI experience.

in this fix i have added a default item which will be show in case the labels list is empty

Fixes #61 

## Type of change
after the fix the label dropdown contains a defalut disabled menuItem if the list is empty

image

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [X] I have performed a self-review of my own code

### CODE REVIEW

File: frontend/src/components/subscribers/index.tsx [hunk 1]
```diff
@@ -206,11 +206,15 @@ export const Subscribers = () => {
...
-              </MenuItem>
-            ))}
+            {!!labels.length ? (
+              labels.map((label) => (
+                <MenuItem key={label.id} value={label.id}>
+                  {label.name}
+                </MenuItem>
+              ))
+            ) : (
+              <MenuItem disabled>No label found</MenuItem>
```
Comment (yassinedorbozgithub) [inline frontend/src/components/subscribers/index.tsx:216]: Hi @JaiSharma1710, can you please use i18n to support French and English translations.
Kindly find bellow the paths to the i18n translation files :
French translations : `frontend/src/i18n/fr/translation.json`
English translations : `frontend/src/i18n/en/translation.json`
I suggestion to use the template feature to generate generic translations

Comment (yassinedorbozgithub) [inline frontend/src/components/subscribers/index.tsx:216]: Hi @JaiSharma1710, can you please use i18n to support French and English translations.
Kindly find bellow the paths to the i18n translation files :
French translations : `frontend/src/i18n/fr/translation.json`
English translations : `frontend/src/i18n/en/translation.json`
I suggestion to use the template feature to generate generic translations

**Review by yassinedorbozgithub — CHANGES_REQUESTED** [review]

File: frontend/src/components/subscribers/index.tsx (hunk 1 above)
Comment (JaiSharma1710) [reply frontend/src/components/subscribers/index.tsx]: updated the pr please check

File: frontend/src/i18n/fr/translation.json [hunk 2]
```diff
@@ -100,7 +100,8 @@
     "invalid_max_fallback_attempt_limit": "La limite des tentatives de secours doit être un nombre positif.",
     "regex_is_invalid": "Le regex est invalide",
     "attachment_not_found": "La pièce jointe est introuvable",
-    "title_length_exceeded": "Vous avez atteint la longueur maximale"
+    "title_length_exceeded": "Vous avez atteint la longueur maximale",
+    "no_label_found": "Aucun label trouvé"
```
Comment (yassinedorbozgithub) [inline frontend/src/i18n/fr/translation.json:104]:
```suggestion
    "no_label_found": "Aucune étiquette trouvée"
```
We can notice that the label is named "étiquette" in the other frensh translations
For example :
- "messge.labels_are_required": "Vous devez ajouter au moins une **étiquette**"
- "message.labels": "**Étiquettes**"

**Review by yassinedorbozgithub — CHANGES_REQUESTED** [review]

File: frontend/src/components/subscribers/index.tsx (hunk 1 above)
Comment (yassinedorbozgithub) [reply frontend/src/components/subscribers/index.tsx]: Hi @JaiSharma1710, things looks much better :rocket: 
please check this small comment https://github.com/Hexastack/Hexabot/pull/62#pullrequestreview-2320482717

Comment (JaiSharma1710) [reply frontend/src/components/subscribers/index.tsx]: ohh i don't know french that's why the issue was there. Anyways i have fixed and pushed the changes

Comment (yassinedorbozgithub) [reply frontend/src/components/subscribers/index.tsx]: No worries @JaiSharma1710.
Thank you for your valuable contribution :tada:

**Review by yassinedorbozgithub — APPROVED** [review]
LGTM :+1:

**Review by medtaher123 — APPROVED** [review]

**Review by marrouchi — APPROVED** [review]

**Comment by marrouchi** [comment]
Thank you very much @JaiSharma1710  🙇

**Pull request merged**